### PR TITLE
[CID-1376] - Add optional tenant filter to List Effective Roles call

### PR DESCRIPTION
**Pull request by kidster**

### CODE REVIEW

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+List effective roles assigned to a user
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+    GET /v2.0/users/{userId}/RAX-AUTH/roles
+
+This operation returns the list of all roles that are assigned to the user with
+the specified user Id. If you don't know the user ID, use the
+:ref:`Get user by Id <get-user-by-id-v2.0>` operation to find it.
```
Comment (dimtruck) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: I think you mean by name?  If you don't know the id, this call would be useless I think

Comment (kidster) [reply api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: Yep, that would make perfect sense. I'll make the change.

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+       roles
+
+The ``sourceId`` identifies the id of the ``sourceType``. For example, the
+``sourceId`` for the source with a ``USER`` source type is the ID for the
+user to which the role was directly assigned, whereas the ``sourceId`` for a
+``USERGROUP`` type would be the ID for the user group.
+
+Assignment type
+^^^^^^^^^^^^^^^
+Role assignments are classified in to three ways based on how the tenants for
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: Role assignments are classified based on how the tenants, for which those assignments apply, are determined. The classifications are shown in the following table:

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+which those assignments apply are determined.
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: delete this line (it's included in previous line)

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+     - Applies to all tenants across all domains within the RCN for which the
+       RCN role applies.
+
+--------
+Examples
+--------
+**Generic Example Response**
+
+This example is contrived to show the different ``sourceTypes`` in a single
+example, and is not meant to represent an real world example.
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: ... and it is not meant to represent a real-world ...

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+                  }
+                ]
+            }
+        ]
+    }
+ }
+
+**Across Domains Assignment Example Response**
+
+Assume:
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: This example depends on the following assumptions:

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+
+- d1t1 and d1t2 are tenants within the same domain (Domain 1).
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: for all three assumptions, put the tenants ``d1t1``, ``d1t2``, and ``d2t1`` between double backticks

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+                  }
+                ]
+            }
+        ]
+    }
+ }
+
+**RCN Role Example Response**
+
+Assume:
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: This example depends on the following assumptions:

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+
+- The user's RCN contains the domain Domain 1 with the d1t1 and d1t2 tenants.
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: for all three assumptions, put the tenants ``d1t1``, ``d1t2``, and ``d2t1`` between double backticks

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+
+**User without Tenants Example Response**
+
+A user could be assigned a role that doesn't apply to *any* current tenants for
+the user. For example, the user may have only DOMAIN roles on a domain with no
+tenants. The user could also be assigned an RCN role that doesn't match any
+tenant within the user's RCN. The service returns the role, but shows that that
+source doesn't apply to any tenants.
+
+Assume:
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: This example depends on the following assumptions:

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+
+- The user's domain does not contain any tenants
+- The user has the identity:user-admin role
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: put ``identity:user-admin`` between double backticks

Also, add a period after the final word (role).

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+
+A user could be assigned a role that doesn't apply to *any* current tenants for
+the user. For example, the user may have only DOMAIN roles on a domain with no
+tenants. The user could also be assigned an RCN role that doesn't match any
+tenant within the user's RCN. The service returns the role, but shows that that
+source doesn't apply to any tenants.
+
+Assume:
+
+- The user's domain does not contain any tenants
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: The user's domain does not contain any tenants.

File: api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst [hunk 12]
```diff
@@ -0,0 +1,404 @@
...
+   GET /v2.0/users/123456/RAX-AUTH/roles HTTP/1.1
+   Host: identity.api.rackspacecloud.com
+   X-Auth-Token: 771dd18634734a46b49f2af620112345
+   Accept: application/json
+   Content-type: application/json
+
+Response
+--------
+
+The response groups returned data by role. Each role is shown as a
```
Comment (catlook) [inline api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst]: ... groups return data ...

**Pull request merged**